### PR TITLE
Adding ProxySQL Diamond Collector

### PR DESCRIPTION
**Pull request by ClareCat**
Added ProxySQL Diamond Collector. Also added tox so the tests could be run with package dependencies.

### CODE REVIEW

**Comment by ClareCat** [comment]
Ah crap. I forgot to add tox to travis. Will do after lunch.

File: src/diamond/collectors/proxysql/proxysqlstat.py [hunk 1]
```diff
@@ -0,0 +1,243 @@
+# coding=utf-8
+
+import diamond.collector
+from diamond.collector import str_to_bool
```
Comment (dichiarafrancesco) [inline src/diamond/collectors/proxysql/proxysqlstat.py]: remove this since never used.

File: src/diamond/collectors/proxysql/proxysqlstat.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+                if metric_name not in self.MYSQL_STATS_COMMANDS_COUNTERS:
+                    metric_value = self.derivative(cmd_nickname + metric_name,
+                                                   metric_value)
+                if (('publish' not in self.config or
+                     metric_name in self.config['publish'])):
+                    self.publish(cmd_nickname + metric_name, metric_value)
+        self.dimensions = {}
+
+    def collect(self):
+        for host in self.config['hosts']:
```
Comment (dichiarafrancesco) [inline src/diamond/collectors/proxysql/proxysqlstat.py]: should you check if MySQLdb got imported before doing anything else?

File: src/diamond/collectors/proxysql/proxysqlstat.py [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+        for cmd_name in metrics[key]:
+            self.dimensions = {'command': cmd_name}
+            cmd_nickname = nickname + cmd_name + '.'
+            for metric_name in metrics[key][cmd_name]:
+                metric_value = metrics[key][cmd_name][metric_name]
+
+                if type(metric_value) is not float:
+                    continue
+
+                if metric_name not in self.MYSQL_STATS_COMMANDS_COUNTERS:
```
Comment (dichiarafrancesco) [inline src/diamond/collectors/proxysql/proxysqlstat.py]: are those counters? if so, why don't we emit by using publish_counter()?

File: src/diamond/collectors/proxysql/proxysqlstat.py [hunk 4]
```diff
@@ -51,11 +50,6 @@ class ProxySQLCollector(diamond.collector.Collector):
         'mysql_session_internal_bytes',
     ]
 
-    MYSQL_STATS_COMMANDS_COUNTERS = [
```
Comment (dichiarafrancesco) [inline src/diamond/collectors/proxysql/proxysqlstat.py]: don't you need these metrics?

Comment (dichiarafrancesco) [reply src/diamond/collectors/proxysql/proxysqlstat.py]: ack

**Pull request merged**